In [2]:
import pandas as pd
import numpy as np

In [181]:
titanic_df=pd.read_csv("train.csv")
tmissing_df=pd.read_csv("train_missing.csv")

titanic datasını okuyunuz(<b>train.csv</b>). Ödevin geri kalanında dataframe gösterimlerini <b>display()</b> fonksiyonunu kullanarak yapınız.<br> Örn: display(titanic_df)

In [331]:
display(titanic_df)

PassengerId  Survived  Pclass  \
0              1         0       3   
1              2         1       1   
2              3         1       3   
3              4         1       1   
4              5         0       3   
..           ...       ...     ...   
886          887         0       2   
887          888         1       1   
888          889         0       3   
889          890         1       1   
890          891         0       3   

                                                  Name     Sex   Age  SibSp  \
0                              Braund, Mr. Owen Harris    male  22.0      1   
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                               Heikkinen, Miss. Laina  female  26.0      0   
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                             Allen, Mr. William Henry    male  35.0      0   
..                                                 ...     ...   ...    ...   
886                              Montvila, Rev. Juozas    male  27.0      0   
887                       Graham, Miss. Margaret Edith  female  19.0      0   
888           Johnston, Miss. Catherine Helen "Carrie"  female   NaN      1   
889                              Behr, Mr. Karl Howell    male  26.0      0   
890                                Dooley, Mr. Patrick    male  32.0      0   

     Parch            Ticket     Fare Cabin Embarked  
0        0         A/5 21171   7.2500   NaN        S  
1        0          PC 17599  71.2833   C85        C  
2        0  STON/O2. 3101282   7.9250   NaN        S  
3        0            113803  53.1000  C123        S  
4        0            373450   8.0500   NaN        S  
..     ...               ...      ...   ...      ...  
886      0            211536  13.0000   NaN        S  
887      0            112053  30.0000   B42        S  
888      2        W./C. 6607  23.4500   NaN        S  
889      0            111369  30.0000  C148        C  
890      0            370376   7.7500   NaN        Q  

[891 rows x 12 columns]

# Soru 1:

Kazada ölenlerin yaş ortalamasını bulunuz

In [315]:
titanic_df.query("Survived==0.0").Age.mean()

30.62617924528302

Kazada ölenlerin bilet fiyatlarının ortalamasını ve medyanını bulunuz

In [332]:
titanic_df.query("Survived==0.0").Fare.mean()

22.117886885245877

In [333]:
titanic_df.query("Survived==0.0").Fare.median()

10.5

Kazada ölen erkeklerin yaş ortalamasını bulunuz


In [334]:
titanic_df.query("Survived==0.0 and Sex=='male'").Age.mean()

31.618055555555557

Kazada ölen Kadınların yaş ortalamasını bulunuz

# Soru 2:

Kazadan kurtulanların yaş ortalamasını bulunuz

In [335]:
titanic_df.query("Survived==1.0").Age.mean()

28.343689655172415

Kazadan kurtulanların bilet fiyatlarının ortalamasını bulunuz


In [336]:
titanic_df.query("Survived==1.0").Fare.mean()

48.39540760233917

Kazadan kurtulan toplam kişi sayısını bulunuz

In [337]:
titanic_df.query("Survived==1.0").Survived.count()

342

Kazadan kurtulan erkeklerin yaş ortalamasını bulunuz

In [338]:
titanic_df.query("Survived==1.0 and Sex=='male'").Age.mean()

27.276021505376345

Kazadan kurtulan kadınların yaş ortalamasını bulunuz

# Soru 3: 

10 yaşından küçüklerin bilet fiyatlarının medyan değerini bulunuz

In [339]:
titanic_df.query("Age<10").Fare.median()

27.0

10 yaşından büyüklerin bilet fiyatlarının medyan değerini bulunuz

In [340]:
titanic_df.query("Age>10").Fare.median()

13.92915

# Soru 5:

1.sınıf, 2.sınıf ve 3.sınıf yolcuların ölüm oranlarını karşılaştırınız. Bir veya bir kaç cümle ile sonucu <b>yorumlayınız.</b>


In [183]:
dead1=titanic_df.query("Pclass==1").Survived.sum()
sum1=titanic_df.query("Pclass==1").Survived.count()
class_1=100-(dead1/sum1)*100

dead2=titanic_df.query("Pclass==2").Survived.sum()
sum2=titanic_df.query("Pclass==2").Survived.count()
class_2=100 -(dead2/sum2)*100

dead3=titanic_df.query("Pclass==3").Survived.sum()
sum3=titanic_df.query("Pclass==3").Survived.count()
class_3=100-(dead3/sum3)*100

print("1.sınıf yolcuların ölüm oranı %",round(class_1,2))
print("2.sınıf yolcuların ölüm oranı %",round(class_2,2))
print("3.sınıf yolcuların ölüm oranı %",round(class_3,2))

1.sınıf yolcuların ölüm oranı % 37.04
2.sınıf yolcuların ölüm oranı % 52.72
3.sınıf yolcuların ölüm oranı % 75.76


1. sınıf yolcuların ölüm oranı 2. ve 3. sınıf yolculara göre daha düşüktür. En yüksek ölüm oranı ise 3. sınıf yolcularda olmuştur.

1.sınıf, 2.sınıf ve 3.sınıf bilet fiyatlarını karşılaştırınız

In [182]:
fare_1=titanic_df.query("Pclass==1").Fare.describe(include="all")
fare_2=titanic_df.query("Pclass==2").Fare.describe(include="all")
fare_3=titanic_df.query("Pclass==3").Fare.describe(include="all")

fsum_1=titanic_df.query("Pclass==1").Fare.count()
fsum_2=titanic_df.query("Pclass==2").Fare.count()
fsum_3=titanic_df.query("Pclass==3").Fare.count()

print(fare_1,fare_2,fare_3)
print(fsum_1,fsum_2,fsum_3)


count    216.000000
mean      84.154687
std       78.380373
min        0.000000
25%       30.923950
50%       60.287500
75%       93.500000
max      512.329200
Name: Fare, dtype: float64 count    184.000000
mean      20.662183
std       13.417399
min        0.000000
25%       13.000000
50%       14.250000
75%       26.000000
max       73.500000
Name: Fare, dtype: float64 count    491.000000
mean      13.675550
std       11.778142
min        0.000000
25%        7.750000
50%        8.050000
75%       15.500000
max       69.550000
Name: Fare, dtype: float64
216 184 491


1.sınıf yolcuların bilet fiyatı ortalaması:84.15
2.sınıf yolcuların bilet fiyatı ortalaması:20.66
3.sınıf yolcuların bilet fiyatı ortalaması:13.67

Bu verilere bakarak bilet fiyatı ile hayatta kalma oranı arasında doğru orantı vardır diyebiliriz.

# Soru 6:

Kazada ölen kadınların oranı ile erkeklerin oranını karşılaştırınız. Bir veya bir kaç cümle ile sonucu <b>yorumlayınız.</b>

In [327]:
female=titanic_df.query("Survived==0.0 and Sex=='female'").Sex.count()
male=titanic_df.query("Survived==0.0 and Sex=='male'").Sex.count()
died=titanic_df.query("Survived==0.0").Survived.count()

oran1=round((female/died)*100,2)
oran2=round((male/died)*100,2)

print("Kazada ölen kadınların oranı %",str(oran1))
print("Kazada ölen erkeklerin oranı %",str(oran2))


Kazada ölen kadınların oranı % 14.75
Kazada ölen erkeklerin oranı % 85.25


Kaza sırasında kadınların yaşaması için öncelik tanın

# Soru 7:

Kazada ölen erkeklerin ortalama bilet fiyatı ile kurtulan erkeklerin ortalama bilet fiyatını karşılaştırınız. Bir veya bir kaç cümle ile sonucu <b>yorumlayınız.</b>

In [328]:
died=round(titanic_df.query("Sex=='male' and Survived==0").Fare.mean(),2)
live=round(titanic_df.query("Sex=='male' and Survived==1.0").Fare.mean(),2)
print("Kazada ölen erkeklerin ortalama bilet fiyatı : %",str(died))
print("Kazada kurtulan erkeklerin ortalama bilet fiyatı :  %",str(live))

Kazada ölen erkeklerin ortalama bilet fiyatı : % 21.96
Kazada kurtulan erkeklerin ortalama bilet fiyatı :  % 40.82


# Missing Value Doldurma

Bundan sonraki soruları <b>train_missing.csv</b> datasını kullanarak cevaplayınız.

# Soru 8

Age, Embarked ve Fare değişkenlerini mean/median/mode değerlerini kullanarak değişken tipine uygun olacak şekilde doldurunuz.

### Age değerlerinin mean ile doldurulması

In [453]:
tmissing_df.Age.isnull().sum()

177

In [363]:
a1=tmissing_df.Age.mean() 

tmissing_df_1=tmissing_df.copy()
tmissing_df_1.Age=tmissing_df_1.Age.fillna(a1)

tmissing_df_1.Age.isnull().sum()

0

In [356]:
tmissing_df_1.Age.mean() ,tmissing_df.Age.mean()

(29.699117647058763, 29.69911764705882)

### Age değerlerinin median ile doldurulması


In [365]:
a2=tmissing_df.Age.median() 

tmissing_df_2=tmissing_df.copy()
tmissing_df_2.Age=tmissing_df_2.Age.fillna(a2)

tmissing_df_2.Age.isnull().sum()

0

In [366]:
tmissing_df_2.Age.median() ,tmissing_df.Age.median()

(28.0, 28.0)

### Embarked değerlerinin mode ile doldurulması

In [395]:
tmissing_df.Embarked.isnull().sum()

173

In [400]:
b1=tmissing_df.Embarked.mode()[0]

tmissing_df_3=tmissing_df.copy()
tmissing_df_3.Embarked=tmissing_df_3.Embarked.fillna(b1)

tmissing_df_3.Embarked.isnull().sum()

0

In [404]:
tmissing_df_3.Embarked.mode() , tmissing_df.Embarked.mode()[0]

(0    S
 dtype: object,
 'S')

### Fare değerlerinin mean ile doldurulması


In [377]:
tmissing_df.Fare.isnull().sum()

91

In [381]:
b1=tmissing_df.Fare.mean()

tmissing_df_4=tmissing_df.copy()
tmissing_df_4.Fare=tmissing_df_4.Fare.fillna(b1)

tmissing_df_4.Fare.isnull().sum()

0

In [382]:
tmissing_df_4.Fare.mean() ,tmissing_df.Fare.mean()

(31.84482187499989, 31.844821874999965)

### Fare değerlerinin median ile doldurulması


In [23]:
b2=tmissing_df.Fare.median()

tmissing_df_5=tmissing_df.copy()
tmissing_df_5.Fare=tmissing_df_5.Fare.fillna(b2)

tmissing_df_5.Fare.isnull().sum()

0

In [24]:
tmissing_df_5.Fare.median() , tmissing_df.Fare.median()

(14.45625, 14.45625)

# Soru 9

SibSp değişkeni hangi yöntemler ile doldurulabilir, fikir yürütün. Uygun bir yöntem ile doldurun. (SibSp değişkeni kişilerin gemide bulunan kardeş/eş sayısını belirtiyor.)

In [163]:
tmissing_df_3=tmissing_df.copy()

In [174]:
tmissing_df_3.SibSp=tmissing_df_3.SibSp.fillna(tmissing_df_3.SibSp.mode())

In [173]:
tmissing_df_3.SibSp.isnull().sum()

0

# Soru 10

Datayı baştan okuyun. Embarked değişkenini "NAN" şeklinde yeni bir kategorisi olacak şekilde doldurun.

In [4]:
tmissing_df = pd.read_csv("train_missing.csv")
tmissing_df.Embarked = tmissing_df.Embarked.fillna("NAN")

In [17]:
tmissing_df.Embarked.isnull().sum()

0

In [185]:
display(tmissing_df.Embarked.value_counts())

S    529
C    128
Q     61
Name: Embarked, dtype: int64

# Soru 11

1"Fare" değişkeni için "Fare_Missing" adında yeni bir değişken türetip "Fare" değişkenini median ile doldurun.

In [186]:
tmissing_df_2=tmissing_df.copy()
na_indices=tmissing_df.Fare[tmissing_df.Fare.isnull()].index.to_list()

tmissing_df_2.insert(10,"Fare_Missing",0)
tmissing_df_2.loc[na_indices,"Fare_Missing"]=1

display(tmissing_df_2)

PassengerId  Survived  Pclass  \
0              1         0       3   
1              2         1       1   
2              3         1       3   
3              4         1       1   
4              5         0       3   
..           ...       ...     ...   
886          887         0       2   
887          888         1       1   
888          889         0       3   
889          890         1       1   
890          891         0       3   

                                                  Name     Sex   Age  SibSp  \
0                              Braund, Mr. Owen Harris    male  22.0    1.0   
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0    1.0   
2                               Heikkinen, Miss. Laina  female  26.0    0.0   
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0    1.0   
4                             Allen, Mr. William Henry    male  35.0    0.0   
..                                                 ...     ...   ...    ...   
886                              Montvila, Rev. Juozas    male  27.0    0.0   
887                       Graham, Miss. Margaret Edith  female  19.0    0.0   
888           Johnston, Miss. Catherine Helen "Carrie"  female   NaN    1.0   
889                              Behr, Mr. Karl Howell    male  26.0    0.0   
890                                Dooley, Mr. Patrick    male  32.0    0.0   

     Parch            Ticket     Fare  Fare_Missing Cabin Embarked  
0        0         A/5 21171   7.2500             0   NaN        S  
1        0          PC 17599  71.2833             0   C85        C  
2        0  STON/O2. 3101282   7.9250             0   NaN      NaN  
3        0            113803  53.1000             0  C123        S  
4        0            373450   8.0500             0   NaN      NaN  
..     ...               ...      ...           ...   ...      ...  
886      0            211536  13.0000             0   NaN        S  
887      0            112053  30.0000             0   B42        S  
888      2        W./C. 6607      NaN             1   NaN        S  
889      0            111369  30.0000             0  C148      NaN  
890      0            370376   7.7500             0   NaN        Q  

[891 rows x 13 columns]

In [42]:
tmissing_df_2.Fare.median()

14.45625

In [6]:
tmissing_df_2.Fare.fillna(tmissing_df_2.Fare.median(),inplace=True)
display(tmissing_df_2)

PassengerId  Survived  Pclass  \
0              1         0       3   
1              2         1       1   
2              3         1       3   
3              4         1       1   
4              5         0       3   
..           ...       ...     ...   
886          887         0       2   
887          888         1       1   
888          889         0       3   
889          890         1       1   
890          891         0       3   

                                                  Name     Sex   Age  SibSp  \
0                              Braund, Mr. Owen Harris    male  22.0    1.0   
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0    1.0   
2                               Heikkinen, Miss. Laina  female  26.0    0.0   
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0    1.0   
4                             Allen, Mr. William Henry    male  35.0    0.0   
..                                                 ...     ...   ...    ...   
886                              Montvila, Rev. Juozas    male  27.0    0.0   
887                       Graham, Miss. Margaret Edith  female  19.0    0.0   
888           Johnston, Miss. Catherine Helen "Carrie"  female   NaN    1.0   
889                              Behr, Mr. Karl Howell    male  26.0    0.0   
890                                Dooley, Mr. Patrick    male  32.0    0.0   

     Parch            Ticket      Fare  Fare_Missing Cabin Embarked  
0        0         A/5 21171   7.25000             0   NaN        S  
1        0          PC 17599  71.28330             0   C85        C  
2        0  STON/O2. 3101282   7.92500             0   NaN      NAN  
3        0            113803  53.10000             0  C123        S  
4        0            373450   8.05000             0   NaN      NAN  
..     ...               ...       ...           ...   ...      ...  
886      0            211536  13.00000             0   NaN        S  
887      0            112053  30.00000             0   B42        S  
888      2        W./C. 6607  14.45625             1   NaN        S  
889      0            111369  30.00000             0  C148      NAN  
890      0            370376   7.75000             0   NaN        Q  

[891 rows x 13 columns]